src/components/FormType.tsx

```diff
@@ -0,0 +1,123 @@
+import { useContext } from "react"
+import { ModalContext } from "~/state/ModalContext"
+
+type ModalStateSetter = (args_0: boolean) => void
+
+interface GitHubFormTypeProps {
+  modals: [ModalStateSetter, ModalStateSetter];
+}
+
+const FormType = () => {
+  const { eventTrigger, setRulesetModalOpened, setCreateModalOpened } = useContext(ModalContext);
+
+  return (
+    <div>
+      {eventTrigger === "mail" && <MailFormType />}
+      {eventTrigger === "calendar" && <CalendarFormType />}
+      {eventTrigger === "github" && <GitHubFormType modals={[setCreateModalOpened, setRulesetModalOpened]} />}
+    </div>
+  );
+};
+
+
+const MailFormType = () => (
+  <div>
+    this is a mail only form
+  </div>
+)
+
+const CalendarFormType = () => (
+  <div>
+    this is a calendar only form
+  </div>
+)
+const GitHubFormType = ({ modals }: GitHubFormTypeProps) => {
+
+  const [C, RS] = modals
```

Comment:
are single letter variable names idiomatic for working with modals?

Reply:
No this should definitely be more descriptive

Reply:
My only defense is because the modal prop is being passed down in the same component and this project has universally only two modals, it is pretty easy to recognize from the modal destructure that these are the two modals but this can be changed. I honestly thought this wouldn't be a big deal unless we should think of this project in terms of scalability.

Reply:
I can change the destructure name to be more clear but this file is just a rough outline of the form. I ideally want to have an `onSubmit` button to work on any form but in this case, it only works on one form. This means I need to write `onSubmit` func for each form which isn't DRY.

Reply:
> My only defense...

No need to defend yourself in code review :)

> it is pretty easy...

Maybe for you as the individual contributor to this, but anytime you stop to call something "easy", you should reflect on how subjective this is and how much it requires someone to understand the codebase. In my opinion it's absolutely impossible to understand what `[C, RS]` mean and `C(false)` is even more confusing. I have to go to entirely different places in your codebase to piece this information together. In contrast you could have something like

```
setCreateModalState({open: false})
```

which anyone could read and immediately understand

> This means I need to write onSubmit func for each form which isn't DRY

Check out this https://kentcdodds.com/blog/aha-programming. TL;DR it's okay to copy paste

Reply:
> No need to defend yourself in code review :)

never meant like that. I just use it as an expression as whatever it is isn't something I strongly advocate too. What I did was a shortcut since it's not a concern for me in comparison to having an e2e working.

> it's absolutely impossible to understand what [C, RS] mean and C(false) is even more confusing.

`setCreateModalState({open: false})` is probably the best way for anyone to understand what is going on. I don't disagree with that at all and I do think my approach can be improved. This can be fixed very quickly but I think I'm more confused on how much quality should be put into creating the e2e implementation.

Also, i checked out the link and didn't know about WET and AHA. Thanks for sending that

Reply:
> This can be fixed very quickly but I think I'm more confused on how much quality should be put into creating the e2e implementation.

It's okay to take shortcuts / do hacky things, but the code should be readable. With reviews, you should keep in mind that you're expecting someone else to able to read and understand what you did / why you did it / find bugs you might've missed. Your responsibility as a good teammate is to make that easier for whoever has to read the code.

Reply:
> Your responsibility as a good teammate is to make that easier for whoever has to read the code.

This is true as it is one of the first times that I get my code actively reviewed. This is something that I need to be more conscious of. I know exactly what C and RS does but I have all the context so I need to remember how something would read this without context.